pkg/kubelet/kubelet.go

```diff
@@ -2353,6 +2353,18 @@ func (kl *Kubelet) generatePodStatus(pod *api.Pod) (api.PodStatus, error) {
 	podFullName := kubecontainer.GetPodFullName(pod)
 	glog.V(3).Infof("Generating status for %q", podFullName)
 
+	if existingStatus, ok := kl.statusManager.GetPodStatus(pod.UID); ok {
+		// This is a hacky fix to ensure container restart counts increment
+		// monotonically. Normally, we should not modify given pod. In this
+		// case, we check if there are cached status for this pod, and update
+		// the pod so that we update restart count appropriately.
+		// TODO(yujuhong): We will not need to count dead containers every time
+		// once we add the runtime pod cache.
+		// Note that kubelet restarts may still cause temporarily setback of
```

Comment:
Good point. Using labels would make restart count survive through kubelet restarts, if the container doesn't get removed. I looked at all the labels we create, and they seem to be more intrinsic to the container itself (e.g., pod/container name, graceful termination period, etc). Plus I am not sure about introducing dependency on getting the last restart count when creating a new container for the pod, as well as using docker as our checkpoint for status. What do you think?